pkg/service/healthserver/healthserver.go

```diff
@@ -189,6 +191,13 @@ func (h *httpHealthHTTPServerFactory) newHTTPHealthServer(port uint16, svc *Serv
 
 		if err := srv.ListenAndServe(); err != http.ErrServerClosed {
 			svc := srv.loadService()
+			if err.Error() == syscall.EADDRINUSE.Error() {
```

Comment:
Two changes requested on this line.

First, the if statement will never be true.
`err.Error()` will be something like `listen tcp :31659: bind: address already in use`, while `syscall.EADDRINUSE.Error()` will be `address already in use`. The two will never be equal. 

Secondly, we are trying to avoid the deprecated `syscall` package in Cilium (see https://github.com/cilium/cilium/issues/10116). Please use `golang.org/x/sys/unix` instead.